model/aggregation.go

```diff
@@ -181,34 +185,70 @@ func (rc *realCluster) aggregateMetrics(target *InfoType, sources []*InfoType) e
 		}
 	}
 
+	if latestTime.Equal(zeroTime) {
+		return fmt.Errorf("aggregateMetrics called with a zero latestTime argument")
+	}
+
 	// Create a map of []TimePoint as a timeseries accumulator per metric
-	newMetrics := make(map[string][]store.TimePoint)
+	newMetrics := make(map[string][]statstore.TimePoint)
 
 	// Reduce the sources slice with timeseries addition for each metric
 	for _, info := range sources {
-		for key, ts := range info.Metrics {
+		for key, ds := range info.Metrics {
 			_, ok := newMetrics[key]
 			if !ok {
 				// Metric does not exist on target map, create a new timeseries
-				newMetrics[key] = []store.TimePoint{}
+				newMetrics[key] = []statstore.TimePoint{}
 			}
 			// Perform timeseries addition between the accumulator and the current source
-			sourceTS := (*ts).Get(rc.timestamp, zeroTime)
-			newMetrics[key] = addMatchingTimeseries(newMetrics[key], sourceTS)
+			sourceDS := (*ds).Hour.Get(rc.timestamp, zeroTime)
+			newMetrics[key] = addMatchingTimeseries(newMetrics[key], sourceDS)
 		}
 	}
 
-	// Put all the new values in the TimeStores under target
+	// Put all the new values in the DayStores under target
 	for key, tpSlice := range newMetrics {
+		if len(tpSlice) == 0 {
+			continue
+		}
 		_, ok := target.Metrics[key]
 		if !ok {
-			// Metric does not exist on target InfoType, create TimeStore
-			newTS := rc.tsConstructor()
-			target.Metrics[key] = &newTS
+			// Metric does not exist on target InfoType, create DayStore
+			target.Metrics[key] = daystore.NewDayStore(epsilonFromMetric(key), rc.resolution)
+		}
+
+		// Put the added TimeSeries in the corresponding DayStore, in time-ascending order
+		for i := len(tpSlice) - 1; i >= 0; i-- {
+			err := target.Metrics[key].Put(tpSlice[i])
+			if err != nil {
+				return fmt.Errorf("error while performing aggregation: %s", err)
+			}
+		}
+
+		// Put a TimePoint with the latest aggregated value at the latest model resolution.
+		// Causes the DayStore to assume the aggregated metric remained constant until the -
+		// next cluster timestamp.
+		newTP := statstore.TimePoint{
+			Timestamp: latestTime,
+			Value:     tpSlice[0].Value,
+		}
+		err := target.Metrics[key].Put(newTP)
+		if err != nil {
+			return fmt.Errorf("error while performing aggregation: %s", err)
 		}
-		for _, tp := range tpSlice {
-			(*target.Metrics[key]).Put(tp)
+
+	}
+
+	// Set the creation time of the entity to the earliest one that we have found data for.
```

Comment:
Tracking the earliest creation timestamp during aggregation helps identify the uptime for abstract entities.
For example, the creation time of a namespace is the creation time of the first pod that was created in that namespace